### PR TITLE
Database implementation

### PR DESCRIPTION
**Pull request by daniel-lerch**
This pull request contains the database model implementation for the Skynet Server. Concurrency issues may still exist. They will be fixed with the packet handler implementation.

### CODE REVIEW

File: src/SkynetServer.Shared/Model/GroupMemberFlags.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SkynetServer.Model
+{
+    public enum GroupMemberFlags : byte
```
Comment (Twometer) [inline src/SkynetServer.Shared/Model/GroupMemberFlags.cs]: Use the [Flags] attribute

File: src/SkynetServer.Shared/Model/MessageFlags.cs [hunk 2]
```diff
@@ -2,9 +2,9 @@
 using System.Collections.Generic;
 using System.Text;
 
-namespace SkynetServer.Network.Model
+namespace SkynetServer.Model
 {
-   public enum MessageFlags
+    public enum MessageFlags : byte
```
Comment (Twometer) [inline src/SkynetServer.Shared/Model/MessageFlags.cs]: Use [Flags]

File: src/SkynetServer.Shared/Migrations/20190102220945_ModelImplementation.cs [hunk 3]
```diff
@@ -0,0 +1,334 @@
+﻿using System;
```
Comment (Twometer) [inline src/SkynetServer.Shared/Migrations/20190102220945_ModelImplementation.cs]: Are these files generated by VS / EF? If so, we could ignore them

Comment (daniel-lerch) [reply src/SkynetServer.Shared/Migrations/20190102220945_ModelImplementation.cs]: Migrations are generated by a PowerShell script in order to update the database model while keeping existing data. As they are generated manually and are manually edited sometimes, they have to be checked in.

Comment (Twometer) [reply src/SkynetServer.Shared/Migrations/20190102220945_ModelImplementation.cs]: Alright.

**Pull request merged**